runtime/doc/lua.txt

```diff
@@ -968,16 +968,35 @@ paste({lines}, {phase})                                          *vim.paste()*
                     |paste|
 
 schedule_wrap({cb})                                      *vim.schedule_wrap()*
-                Defers callback `cb` until the Nvim API is safe to call.
+                Wraps callback `cb` in a function which will defer the execution
+                of `cb` until the Nvim API is safe to call.
+
+                The user must call the returned function themselves if it is not
+                being used as an argument to a function that will call it for them,
+                such as `vim.fn_defer`.
```

Comment:
It's `defer_fn` -- and this function _itself_ wraps the argument in `schedule_wrap`, so mentioning this here is a bit misleading?

Reply:
Maybe it would be better instead to start with the second example and then refer to `defer_fn` as a simple, more "high-level" alternative that defers without requiring wrapping?

Reply:
A better example would be to use a raw luv callback or like nvim_buf_attach.

Reply:
My mistake, I did not realize it automatically added it's own wrapping (Z_Z). I agree this is a poor example.

`nvim_buf_attach` is probably a good choice because it may be an exact time you want to use `schedule_wrap`. I'm not quite sure how complicated to make the callback.

For now I have left it as the "echomsg" function. I couldn't think of something that was simple enough to be one line, while still requiring schedule_wrap. My API knowledge isn't very deep.